wpilibc/src/main/native/cpp/smartdashboard/Mechanism2d.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include "frc/smartdashboard/Mechanism2d.h"
+
+#include <ios>
+
+#include <wpi/raw_ostream.h>
+
+#include "frc/smartdashboard/SendableBuilder.h"
+
+using namespace frc;
+
+Mechanism2d::Mechanism2d(double width, double height) : m_dims{width, height} {}
+
+MechanismRoot2d* Mechanism2d::GetRoot(const wpi::StringRef name, double x, double y) {
+  if (m_roots.count(name)) {
+    return m_roots[name].get();
+  }
+  const auto& twine = wpi::Twine::Twine(name);
+  const std::unique_ptr<MechanismRoot2d> ptr = std::make_unique<MechanismRoot2d>(twine, x, y);
+  m_roots.try_emplace(name, ptr);
+  ptr->Update(m_table->GetSubTable(name));
+  return ptr.get();
+}
+
+void Mechanism2d::SetBackgroundColor(const Color8Bit& color) {
+  std::string buf;
+  wpi::raw_string_ostream os{buf};
+  os << "#";
+  os << std::hex << (color.red << 16 | color.green << 8 | color.blue);
```

Comment:
It might be more readable here to just use a char buffer here since this is fixed size.  So:
```
char buf[10];
std::snprintf(buf, sizeof(buf), "#%02x%02x%02x", color.red, color.green, color.blue);
```